### PR TITLE
Prevent calling assembly on nil object

### PR DESCRIPTION
**Pull request by korki**

### CODE REVIEW

**Comment by dinesh16** [comment]
:raised_hands: :beers: :beers:

**Comment by korki** [comment]
@dinesh16 
cheers

File: app/models/spree/stock/quantifier_decorator.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@ def initialize(variant)
   end
 
   def can_supply?(required)
-    if @product.assembly?
+    if !@product.nil? && @product.assembly?
```
Comment (forvalho) [inline app/models/spree/stock/quantifier_decorator.rb]: i'd recommend `if @product.present? && @product.assembly?` to keep both assertions positive ;)

Comment (korki) [inline app/models/spree/stock/quantifier_decorator.rb]: It's a 10 characters change and I'll get now 100 comments on it? :P

Comment (korki) [inline app/models/spree/stock/quantifier_decorator.rb]: cheers

Comment (dinesh16) [inline app/models/spree/stock/quantifier_decorator.rb]: :laughing: I agree

**Pull request merged**